### PR TITLE
Fix login error

### PR DESCRIPTION
**Pull request by fabriziosestito**
This PR fixes the login page. 

Before:

On a 401 we were showing a toast with a generic error message.
There was a bug and the toast was shown only once after the first attempt.

After:
We show an invalid credential message and we highlight with a red border the form fields if we get a 401.
We show a toast when any other error occurs.

It also disables the login button when submitting the form.
login

### CODE REVIEW

**Review by CDimonaco — APPROVED** [review]

File: assets/js/state/sagas/user.js [hunk 1]
```diff
@@ -24,7 +24,9 @@ export function* performLogin({ payload: { username, password } }) {
     yield call(storeRefreshToken, refreshToken);
     yield put(setUserAsLogged());
   } catch (error) {
-    yield put(setAuthError({ error: error.message }));
+    yield put(
+      setAuthError({ message: error.message, code: error.response.status })
```
Comment (arbulu89) [inline assets/js/state/sagas/user.js:28]: Could this somehow fails on some other error?
Like network error where we don't have a status code.
Would it go to undefined?

File: assets/js/components/Login.jsx [hunk 2]
```diff
@@ -64,7 +67,10 @@ export default function Login() {
                   name="username"
                   autoComplete="username"
                   required
-                  className="appearance-none block w-full px-3 py-2 border border-gray-300 rounded-md shadow-sm placeholder-gray-400 focus:outline-none focus:ring-jungle-green-500 focus:border-jungle-green-500 sm:text-sm"
+                  className={classNames(
+                    'appearance-none block w-full px-3 py-2 border border-gray-300 rounded-md shadow-sm placeholder-gray-400 focus:outline-none focus:ring-jungle-green-500 focus:border-jungle-green-500 sm:text-sm',
+                    { 'border-red-300': authError && authError.code === 401 }
```
Comment (arbulu89) [inline assets/js/components/Login.jsx:72]: Could we wrap this in a function?
```
const unauthorized = (authError) => authError && authError.code === 401;
```

**Review by arbulu89 — APPROVED** [review]
Great!
I couple of comments, that don't need a new review

File: assets/js/components/Login.jsx (hunk 2 above)
Comment (dottorblaster) [reply assets/js/components/Login.jsx:72]: Maybe just having a boolean `const` defined in the function body is enough, I was thinking about the function as well but I don't know if it's the best

**Review by EMaksy — COMMENTED** [review]
LGTM :D

**Review by nelsonkopliku — APPROVED** [review]

**Review by rtorrero — APPROVED** [review]
Other than what has already been said, LGTM

File: assets/js/state/sagas/user.js (hunk 1 above)
Comment (fabriziosestito) [reply assets/js/state/sagas/user.js]: nice catch!

File: assets/js/components/Login.jsx (hunk 2 above)
Comment (fabriziosestito) [reply assets/js/components/Login.jsx:72]: done

**Pull request merged**